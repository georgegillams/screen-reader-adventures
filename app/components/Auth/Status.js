import React from 'react';
import PropTypes from 'prop-types';

import { Section } from 'gg-components/dist/Typography';

const Status = props => {
  const { success, error } = props;

  if (success || error) {
    return (
      <Section name="STATUS">
        {success && <span>SUCCESS</span>}
        {error && <span>{error}</span>}
      </Section>
    );
  }

  return null;
};

Status.propTypes = {
  children: PropTypes.node.isRequired,
  success: PropTypes.bool,
  error: PropTypes.string,
};

Status.defaultProps = {
  success: false,
  error: null,
};

export default Status;
