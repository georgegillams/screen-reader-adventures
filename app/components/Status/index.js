export { default } from './Status';
export StatusCard from './StatusCard';
export STATUS_TYPES from './StatusTypes';
