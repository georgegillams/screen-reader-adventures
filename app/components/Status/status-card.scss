@import '~bpk-mixins';

@fontface {
  font-family: 'Space Mono', monospace;
  src: url('https://fonts.googleapis.com/css?family=Space Mono');
}

.status-card {
  &__container {
    max-width: 20rem;
    width: 100%;
    position: relative;
  }

  &__status {
    margin-left: calc(50% - 2.5rem);
    z-index: 1000;
    top: 0;
    position: absolute;
    margin-top: -2.5rem;
  }

  &__card {
    margin-top: 2.5rem;
  }

  &__content-item {
    width: 100%;
    display: flex;
    align-items: center;
    margin-top: 1rem;
  }

  &__content-item-status {
    margin-right: 1rem;
  }

  &__content-items {
    margin-top: 1rem;
  }
}
