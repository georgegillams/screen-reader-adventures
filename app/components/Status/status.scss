@import '~bpk-mixins';

@fontface {
  font-family: 'Space Mono', monospace;
  src: url('https://fonts.googleapis.com/css?family=Space Mono');
}

.status {
  &__outer-container {
  }

  &__shadow-container {
    height: 2.5rem;
    overflow: hidden;
    width: 7rem;
    position: absolute;
    z-index: -4;
  }

  &__shadow {
    border-radius: 10rem;
    width: 5rem;
    height: 5rem;
    box-shadow: 0.1rem 0.1rem 0.3rem 0 rgba(37, 32, 31, 0.3);
    z-index: -3;
  }

  &__container {
    border: 0.3rem solid blue;
    width: 2rem;
    height: 2rem;
    border-radius: 10rem;
    display: flex;
    justify-content: center;
    align-items: center;
    color: blue;
    fill: blue;
    background-color: white;
    font-size: 1.5rem;

    &--success {
      border-color: green;
      color: green;
      fill: green;
    }

    &--warn {
      border-color: orange;
      color: orange;
      fill: orange;
    }

    &--error {
      border-color: #e02626;
      color: #e02626;
      fill: #e02626;
    }

    &--large {
      width: 5rem;
      height: 5rem;
      border-width: 0.35rem;
      font-size: 5rem;
    }
  }
}
