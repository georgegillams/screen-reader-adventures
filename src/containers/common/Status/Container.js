import React from 'react';
import PropTypes from 'prop-types';
import { cssModules } from 'gg-components/helpers/cssModules';
import { Subsection } from 'gg-components/Subsection';
import { Paragraph } from 'gg-components/Paragraph';
import PageTitle from 'components/common/PageTitle';
import { DebugObject } from 'gg-components/DebugObject';

import STYLES from './style.scss';

import { getTimeDifference } from 'helpers/common/time';

const getClassName = cssModules(STYLES);

const Status = props => {
  const { appConfig, startedAt, builtAt, nodeEnv, ...rest } = props;

  return (
    <div {...rest}>
      <PageTitle anchor={false} name="Status">
        <img
          className={getClassName('style__component')}
          alt="Build status"
          src={`${appConfig.githubRepoUrl}/workflows/CI/badge.svg`}
        />
        <br />
        <img
          className={getClassName('style__component')}
          alt="Dependency status"
          src={`https://img.shields.io/david/${appConfig.githubRepo}`}
        />
        <Subsection className={getClassName('pages__component')} anchor={false}>
          <Paragraph>
            Built {getTimeDifference(new Date(builtAt * 1000))}
            <br />
            Started {getTimeDifference(new Date(startedAt * 1000))}
          </Paragraph>
        </Subsection>
        <Subsection className={getClassName('pages__component')} anchor={false}>
          <Paragraph>Environment {nodeEnv}</Paragraph>
        </Subsection>
        <Subsection className={getClassName('pages__component')} anchor={false}>
          <DebugObject debugObject={appConfig} debugTitle="App config"></DebugObject>
        </Subsection>
      </PageTitle>
    </div>
  );
};

Status.propTypes = {
  appConfig: PropTypes.object.isRequired,
  startedAt: PropTypes.number.isRequired,
  builtAt: PropTypes.number.isRequired,
  nodeEnv: PropTypes.string.isRequired,
};

export default Status;
