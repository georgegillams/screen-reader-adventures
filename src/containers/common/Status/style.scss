@import '~utils/tokens';

.style {
  &__component {
    margin-bottom: $spacing-base;
  }
}
