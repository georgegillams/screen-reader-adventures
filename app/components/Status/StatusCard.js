import React from 'react';
import PropTypes from 'prop-types';
import Status from './Status';
import ArticleCard, { CARD_LAYOUTS } from 'components/Cards';

import STYLES from './status-card.scss'; import {cssModules} from 'bpk-react-utils';  const getClassName = cssModules(STYLES); // REGEX_REPLACED

const StatusCard = props => {
  const { data, overallStatus, className, ...rest } = props;

  const classNameFinal = [getClassName('status-card__container')];
  if (className) classNameFinal.push(className);

  return (
    <div className={classNameFinal.join(' ')} {...rest}>
      <ArticleCard
        layout={CARD_LAYOUTS.narrowCompact}
        className={getClassName("status-card__card")}
      >
        <div className={getClassName("status-card__content-items")}>
          {data &&
            data.map(dataItem => (
              <div className={getClassName("status-card__content-item")}>
                <Status
                  type={dataItem.status}
                  className={getClassName("status-card__content-item-status")}
                />
                {dataItem.item}
                <br />
                <br />
              </div>
            ))}
        </div>
      </ArticleCard>
      <Status
        shadow
        type={overallStatus}
        large
        className={getClassName("status-card__status")}
      />
    </div>
  );
};

StatusCard.propTypes = {
  data: PropTypes.arrayOf(PropTypes.object).isRequired,
  overallStatus: PropTypes.string.isRequired,
  className: PropTypes.string,
};

StatusCard.defaultProps = {
  className: null,
};

export default StatusCard;
