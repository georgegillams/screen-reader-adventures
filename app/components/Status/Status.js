import React from 'react';
import PropTypes from 'prop-types';
import { TextLink } from 'components/Typography';
import STATUS_TYPES from './StatusTypes';

import STYLES from './status.scss'; import {cssModules} from 'bpk-react-utils';  const getClassName = cssModules(STYLES); // REGEX_REPLACED

const Status = props => {
  const { shadow, type, large, className, ...rest } = props;

  const classNameFinal = [getClassName('status__outer-container')];
  if (className) classNameFinal.push(className);

  const statusClassName = [getClassName('status__container')];
  if (large) {
    statusClassName.push(getClassName('status__container--large'));
  }
  if (type === STATUS_TYPES.SUCCESS) {
    statusClassName.push(getClassName('status__container--success'));
  }
  if (type === STATUS_TYPES.WARN) {
    statusClassName.push(getClassName('status__container--warn'));
  }
  if (type === STATUS_TYPES.ERROR) {
    statusClassName.push(getClassName('status__container--error'));
  }

  return (
    <div className={classNameFinal.join(' ')} {...rest}>
      {large && shadow && (
        <div className={getClassName("status__shadow-container")}>
          <div className={getClassName("status__shadow")} />
        </div>
      )}
      <div className={statusClassName.join(' ')}>
        {type === STATUS_TYPES.SUCCESS && '✔'}
        {type === STATUS_TYPES.WARN && '!'}
        {type === STATUS_TYPES.ERROR && large && '✘'}
      </div>
    </div>
  );
};

Status.propTypes = {
  type: PropTypes.string.isRequired,
  large: PropTypes.bool,
  shadow: PropTypes.bool,
  className: PropTypes.string,
};

Status.defaultProps = {
  shadow: false,
  large: false,
  className: null,
};

export default Status;
