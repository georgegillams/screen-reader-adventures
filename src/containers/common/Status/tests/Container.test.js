import React from 'react';
import { render } from '@testing-library/react';

import Status from '../Container';

const testAppConfig = {
  githubRepo: 'GITHUB_REPO',
  githubRepoUrl: 'GITHUB_REPO_URL',
};

describe('<Status />', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Status appConfig={testAppConfig} startedAt={12375223} builtAt={12355123} nodeEnv="test-node-env" />
    );

    expect(container).toMatchSnapshot();
  });
});
