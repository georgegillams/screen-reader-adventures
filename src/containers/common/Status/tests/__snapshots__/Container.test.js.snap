// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Status /> should render correctly 1`] = `
<div>
  <div>
    <h1
      class="page-title__heading"
    >
      Status
    </h1>
    <img
      alt="Build status"
      class="style__component"
      src="GITHUB_REPO_URL/workflows/CI/badge.svg"
    />
    <br />
    <img
      alt="Dependency status"
      class="style__component"
      src="https://img.shields.io/david/GITHUB_REPO"
    />
    <div
      class="subsection__outer pages__component"
    >
      <span
        class="paragraph"
      >
        Built 
        almost 51 years ago
        <br />
        Started 
        almost 51 years ago
      </span>
    </div>
    <div
      class="subsection__outer pages__component"
    >
      <span
        class="paragraph"
      >
        Environment 
        test-node-env
      </span>
    </div>
    <div
      class="subsection__outer pages__component"
    >
      <div
        class="section__outer debug-object__main"
      >
        <h2
          class="section__heading"
          id="app-config"
        >
          App config
        </h2>
        <div
          role="button"
          style="margin-left: 0rem;"
          tabindex="0"
        >
          ▶️ top-level:
        </div>
      </div>
    </div>
  </div>
</div>
`;
